src/mqtt.c

```diff
@@ -29,8 +29,144 @@ static void mg_send_u16(struct mg_connection *c, uint16_t value) {
   mg_send(c, &value, sizeof(value));
 }
 
+static uint8_t compute_variable_length_size(uint32_t length) {
+  uint8_t bytes_needed = 0;
+  do {
+    bytes_needed++;
+    length /= 0x80;
+  } while (length > 0);
+
+  return bytes_needed;
+}
+
+static uint32_t decode_variable_length(const char *buf, uint32_t *bytes_consumed) {
+    const uint8_t *p = (const uint8_t *)buf;
+    uint32_t value = 0;
+    uint32_t multiplier = 1;
+    uint8_t encoded_byte;
+    uint32_t offset;
+
+    for (offset = 0; offset < 4; offset++) {
+        encoded_byte = p[offset];
+        value += (encoded_byte & 0x7F) * multiplier;
+        multiplier *= 128;
+
+        if (!(encoded_byte & 0x80))
+            break;
+    }
+
+    if (bytes_consumed != NULL)
```

Comment:
Please use LLVM's clang-format for styling the code. Use "file" mechanism. We have a .clang-format config file in the repo root.